### ISSUE TITLE
Dropout layer works wrongly when rate = 1.0

### ISSUE THREAD

**Issue by DHKLeung** [keras-team/keras#9141]
just asked in [stackoverflow](https://stackoverflow.com/questions/48355700/dropout-layer-in-keras-is-not-working)
`input0 = keras.layers.Input((32, 32, 3), name='Input0')`
`flatten = keras.layers.Flatten(name='Flatten')(input0)`
`relu1 = keras.layers.Dense(256, activation='relu', name='ReLU1')(flatten)`
`dropout = keras.layers.Dropout(1., name='Dropout')(relu1)`
`softmax2 = keras.layers.Dense(10, activation='softmax', name='Softmax2')(dropout)`
`model = keras.models.Model(inputs=input0, outputs=softmax2, name='cifar')`
I found that dropout works wrongly so i write a simple NN to test it.
when i set the dropout rate = 1.0
it should drop all the nodes from the previous layer and thus the output layer should remain the same from the initialization.
That is, the performance throughout epochs should be the same
however it still work although i drop all the nodes

**Comment by fchollet** [comment]
`rate=1.` is an invalid value and thus results in no dropout being applied.

Values outside of the ]0, 1[ range map to 0.
